include/swift/AST/Decl.h

```diff
@@ -136,6 +136,7 @@ enum class DescriptiveDeclKind : uint8_t {
   GenericStruct,
   GenericClass,
   GenericType,
+  ExtendedType,
```

Comment:
I'm not sure about this unification. This isn't a kind of declaration any more; its saying something about how the declaration is used.

Reply:
Yeah, true. I was not sure either and was curious about your opinion. Let's reset.